src/Microsoft.AspNet.JsonPatch/TypedJsonPatchDocument.cs

```diff
@@ -357,13 +357,17 @@ public void ApplyTo(T objectToApplyTo)
             ApplyTo(objectToApplyTo, new ObjectAdapter<T>(ContractResolver));
         }
 
-        public void ApplyTo(T objectToApplyTo, IObjectAdapter<T> adapter)
+        public void ApplyTo(T objectToApplyTo, Action<string> action)
         {
+            ApplyTo(objectToApplyTo, new ObjectAdapter<T>(ContractResolver), action);
+        }
 
+        public void ApplyTo(T objectToApplyTo, IObjectAdapter<T> adapter, Action<string> action = null)
```

Comment:
Should we have another API where we take a prefix to use for model-state? (Take a look at how input formatters add errors).

Would it be possible to find all/most of the errors during deserialization? Worth thinking about?